torch/csrc/Exceptions.h

```diff
@@ -75,6 +75,9 @@ inline void PyErr_SetString(PyObject* type, const std::string& message) {
   _CATCH_GENERIC_ERROR(                                                       \
       NotImplementedError, PyExc_NotImplementedError, retstmnt)               \
   _CATCH_GENERIC_ERROR(SyntaxError, PyExc_SyntaxError, retstmnt)              \
+  _CATCH_GENERIC_ERROR(DeviceAssertError, PyExc_AssertionError, retstmnt)     \
```

Comment:
Could we please use a custom exception type on the Python side? It's not super-common to handle the built-in `AssertionError` and `MemoryError` in Python code, i.e., it's rare to see `try.. except AssertionError`. This also gives us the flexibility to add more attributes such as the result of `cudaGetLastError`